File: 03- Subprogramas y recursividad/recursividad/java/Fibbonacci.java
/**
 * Descripción: Ejecucion de SistemaVentas.java.
 * Autor: Santiago Nicolás De la mora Núñez
 * Fecha: 05/03/2026
 */

import javax.swing.JOptionPane;

public class Fibbonacci {
    public static void main(String[] args) {
        int numero = 4;

        JOptionPane.showMessageDialog(
                null,
                "Fibbonacci de " + numero + " es " + calcularFibbonacci(numero),
                "Serie Fibbonnacci",
                JOptionPane.PLAIN_MESSAGE
        );
    }

    private static int calcularFibbonacci(int numero) {
        if (numero == 0) {
            return 0;
        } else if (numero == 1) {
            return 1;
        } else {
            return calcularFibbonacci(numero - 1) + calcularFibbonacci(numero - 2);
        }
    }
}

File: 03- Subprogramas y recursividad/recursividad/python/fibbonacci.py
"""
Descripción: Ejecucion de SistemaVentas.java.
Autor: Santiago Nicolás De la mora Núñez
Fecha: 10/04/2026
"""

def main():
    numero = 4
    print(f"Serie Fibbonacci de {numero} es {calcular_fibbonacci(numero)}")

def calcular_fibbonacci(numero):
    if numero == 0:
        return 0
    elif numero == 1:
        return 1
    else:
        return calcular_fibbonacci(numero - 1) + calcular_fibbonacci(numero - 2)

# Ejecuta el programa
if __name__ == '__main__':
    main()